lib/seeder.js

```diff
@@ -0,0 +1,115 @@
+module.exports = Seeder
+
+const WebTorrent = require('webtorrent')
+const IdbChunkStore = require('indexeddb-chunk-store')
+const TabElect = require('tab-elect')
+const path = require('path')
+
+function Seeder () {
+  if (typeof window === 'undefined') throw new Error('must be called in a wep page')
+
+  this.destroyed = false
+  this._seeds = {}
+  this._webtorrent = null
+
+  this._tabElect = new TabElect('planktos')
+  this._tabElect.on('elected', this._onElected.bind(this))
+  this._tabElect.on('deposed', this._onDeposed.bind(this))
+}
+
+Seeder.prototype.add = function (snapshot) {
+  if (this.destroyed) throw new Error('Seeder has been destroyed')
+  if (snapshot.hash in this._seeds) return
+
+  let listener = this._prioritize.bind(this, snapshot)
+
+  this._seeds[snapshot.hash] = {
+    snapshot: snapshot,
+    listener: listener
+  }
+
+  if (this._tabElect.isLeader) this._onElected()
```

Comment:
Just so that any snapshot added after that tab was elected starts being downloaded when it's added and not the next time when a leader is elected. This will be more clear in the commit I'm about to push.